packages/gatsby-theme-aio/src/components/Carousel/index.js

```diff
@@ -0,0 +1,228 @@
+  slideTheme,
+  bulletActiveClass = 'swiper-pagination-bullet-active',
+  bulletClass = 'swiper-pagination-bullet',
+  ...props
+}) => {
+  const propKeys = Object.keys(props);
+  let carouselProps = propKeys.filter((key) => key.startsWith('image'));
+
+  carouselProps = carouselProps.map((data, index) => {
+    return {
+      image: props[data],
+      heading: props[`heading${index}`],
+      buttons: props[`buttons${index}`]
+    };
+  });
+
+  const textKeys = props.slots.split(',').filter((key) => key.trim().startsWith('text'));
+
+  const backgroundColor = `background-color: var(--spectrum-global-color-gray-${slideTheme === 'light' ? '50' : ''});`;
+
+  return (
+    <section
+      className={classNames(className, `spectrum--${theme}`)}
+      css={css`
+        background: var(--spectrum-global-color-gray-100);
+        padding: var(--spectrum-global-dimension-size-600) 0 var(--spectrum-global-dimension-size-200) 0;
+      `}>
+      <div
+        css={css`
+          max-width: calc(${layoutColumns(12)});
+          margin: auto;
+          @media screen and (max-width: ${TABLET_SCREEN_WIDTH}) {
+            padding-bottom: 0;
+            margin-top: 0;
+            max-width: calc(${layoutColumns(6)});
+          }
+          @media screen and (max-width: ${MOBILE_SCREEN_WIDTH}) {
+            max-width: calc(${layoutColumns(3)});
+          }
+        `}>
+        <Swiper
+          speed={swiperSpeed}
+          slidesPerView={'auto'}
+          autoplay={{
+            delay
+          }}
+          pagination={{
+            bulletActiveClass,
+            bulletClass,
+            clickable: true
+          }}
+          navigation={{
+            nextEl: '.swiper-button-next',
+            prevEl: '.swiper-button-prev'
+          }}>
+          {carouselProps.map((data, index) => {
+            return (
+              <SwiperSlide key={index}>
+                <SwiperContent
+                  textKeys={textKeys}
+                  heading={data.heading}
+                  image={data.image}
+                  imageStyle={imageStyle}
+                  buttons={data.buttons}
+                  props={props}
+                  backgroundColor={backgroundColor}
+                  index={index}
+                  slideTheme={slideTheme}
+                  theme={theme}
+                />
+                {enableNavigation ? (
+                  <>
+                    <div className="swiper-button-prev"></div>
+                    <div className="swiper-button-next" />
+                  </>
+                ) : null}
+              </SwiperSlide>
+            );
+          })}
+        </Swiper>
+      </div>
+    </section>
+  );
+};
+
+Carousel.propTypes = {
+  className: PropTypes.string,
+  theme: PropTypes.oneOf(['light', 'dark', 'lightest']),
+  imageStyle: PropTypes.string,
+  swiperSpeed: PropTypes.number,
+  delay: PropTypes.number,
+  enableNavigation: PropTypes.bool,
+  slideTheme: PropTypes.oneOf(['light', 'dark']),
```

Comment:
What's the difference between slideTheme and theme ? Is slideTheme necessary ?

Reply:
![image](https://user-images.githubusercontent.com/532401/135074933-0c7c35a1-0229-4ba5-8ac3-f8bf041a09f6.png)

Theme is used for whole section.
slideTheme is used for swiper Content section like in above image. Its an optional props if we don't send this props It will take theme props here.